### ISSUE TITLE
Neo4j 2.2.1 broken HEAD and LAST functions

### ISSUE THREAD

**Issue by denistoikka** [neo4j/neo4j#4522]
Simple example:

```
CREATE (:A)-[:AB]->(:B)-[:BC]->(:C)
```

then

```
MATCH (:A)-[r*]->(:C) return LAST(r)
```

returns error 

```
Expected `r` to be a Collection<Any> but it was a Relationship
```

In Neo4j 2.2.0 it works correctly and returns last relation in path.

**Comment by danielcor** [comment]
I don't think this issue is limited to HEAD.

I get the same message with:

MATCH (p) WHERE ID(p) IN [ 199505, 281153, 287792, 287793]
                            WITH p 
                            MATCH path=((p)-[r*1..1]-(o))
                            with path,p,o,r,extract(x in r | x.relscore) as scores, length(path) as len
                            RETURN p,o,r,reduce(res=0, x in scores | res +x) as totalscore
                            ORDER BY totalscore LIMIT 100 

I haven't yet reduced it to a simple example.   If I prefix this with CYPHER 2.1, then it works using the older parser.

**Comment by petraselmer** [comment]
Hi all

I just wanted to let you know that the bug fix for this shall be part of the 2.2.2 milestone release, in about a month's time.

**Comment by petraselmer** [comment]
Hi all

Neo4j 2.2.2 - a maintenance release - is now available for download; this resolves the bug.
